homeassistant/components/generic_thermostat/climate.py

```diff
@@ -109,36 +112,68 @@
 ).extend({vol.Optional(v): vol.Coerce(float) for (k, v) in CONF_PRESETS.items()})
 
 
+# Deprecated in Home Assistant 2023.2
 async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
     async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the generic thermostat platform."""
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
+        )
+    )
+
+    _LOGGER.warning(
+        "Configuration of the Generic Thermostat integration in YAML is deprecated and "
+        "will be removed in a 2023.2; Your existing configuration "
+        "has been imported into the UI automatically and can be safely removed "
+        "from your configuration.yaml file"
+    )
```

Comment:
This integration does not connect to a device or service itself, not does it do anything else as defined under ADR-0010. It does, however, consume existing entities or devices in Home Assistant.

This makes this a helper (and not an integration), IMHO. As a helper, we should therefore support both YAML and UI in this case as well.

Reply:
Do we have a concept of a virtual device BTW?

Reply:
I do not understand that question. Could you add some context?

Reply:
Well, that's probably a topic for architecture discussion.